src/admin/bundles/bundles.factory.js

```diff
@@ -18,6 +18,55 @@
                 details: {method: 'GET', params: {action: 'detail'}}
             }
         );
+
+        Bundle.prototype.getVersion = function(){
+            return this.version.major+"."+this.version.minor+"."+this.version.micro+"-"+this.version.qualifier;
+        };
+        Bundle.prototype.getIconURL = function(){
+            return ServerService.formatURL("module/server/module/icon?bundleId=" + this.bundleId);
+        };
+        Bundle.prototype.isActive = function(){
+            return this.state === "ACTIVE";
+        };
+
+
+        function callbackSuccess(){
+            LoadingModal.close();
+            $rootScope.$broadcast('motech.refresh');
+        }
+
+        Bundle.prototype.start = function(){
+            LoadingModal.open();
+            this.$start(callbackSuccess);
+        };
+        Bundle.prototype.stop = function(){
+            MotechConfirm(i18nService.getMessage('admin.bundles.stateChangeWarning')) //'admin.bundles.stop' (stop button text)
+            .then(function(){
+                LoadingModal.open();
+                this.$stop(callbackSuccess);
+            });
+        };
+        Bundle.prototype.restart = function(){
+            LoadingModal.open();
+            this.$restart(callbackSuccess);
+        };
+        
+        Bundle.prototype.uninstallBundle = function(){
+            MotechConfirm(i18nService.getMessage('admin.bundles.stateChangeWarning')) // 'admin.remove' (remove button text)
```

Comment:
I'm a bit confused here, why is there the same message for stopping and uninstalling bundle as comments in lines 43 and 55 suggest that they should be different?